evaluation/scripts/profile-all.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+# This file is part of gobra-libs which is released under the MIT license.
+# See LICENSE or go to https://github.com/viperproject/gobra-libs/blob/main/LICENSE
+# for full license details.
+
+# DEPENDENCIES (plot.py):
+# - pandas
+# - numpy
+# - seaborn
+# - matplotlib
+# In case you forgot to install the dependencies before running the script, you
+# do not need to rerun this script (and in particular profiling); instead, simply use
+# the command for plotting found below.
+
+# USAGE:
+# Switch into the same directory as profile-all.sh, and run the script.
+#
+# In case you are using a laptop, make sure that power settings are set
+# properly:
+# - don't run from battery
+# - make sure the correct profile is set, i.e. not the battery profile
+#   (e.g., in TLP)
+# - make sure the laptop will not suspend
+#   - check packages that may interfere
+#   - use systemd-inhibit
+
+ITERATIONS=30
+
+# Set the path to silicon.sh
+SILICON_PATH=
+# Set the path to the Z3 binary
+# Tested Versions: 4.8.7 (newer versions may not work; produce errors and different output) 
+Z3_PATH=
+# Set the path to the Gobra jar
+GOBRA_PATH=
```

Comment:
Those are good points, thanks for the feedback. I addressed it.

In case it doesn't take too long, would you be willing to run the script and see whether it appears to work (i.e. doesn't crash somehow, and produces csvs/plots)? I am currently having some issues with my machine.

Reply:
> instead of defining Z3_PATH, we can just reuse the variable Z3_EXE which is needed for silicon and carbon anyway

The script requires old versions of Z3, so I think it would be better if these are different variables.

> on Mac, by default, there is no executable called python, but there is one called python3. Maybe using python3 would be more generally supported

On GNU Guix it is the same, but I wasn't sure whether it is a quirk of it. I will change it.

> After running it on my machine with all occurrences of python replaced by python3, I get the output as shown in [output.txt](https://github.com/viperproject/gobra-libs/files/15112962/output.txt)

I came across this problem myself too sometimes. Usually it is fixed by pulling and building Silicon again. I will add a note to the README.

> maybe we should put in the readme the commands that we need to install the necessary dependencies

I think it is a good idea to put the dependencies also into the README. Since installing packages differs from distribution to distribution (pip, for example, refuses to install packages on some Linux distributions, as it doesn't want to mess with the system packages), I will add a generic note.

Reply:
> The script requires old versions of Z3, so I think it would be better if these are different variables.

I'm not sure if that is true. In principle, the script should use the version of z3 that is used by the tools for the results to be meaningful. Also, we can always change the value of `Z3_EXE` for the current shell.

Reply:
I agree that that's the way it should be; however, the issue is described here https://github.com/viperproject/silicon/issues/786.

For the script to be able to use newer versions of Z3, the following things probably need to be done:
- fix `Invalid quantifier instances line from prover:`
- find documentation for the columns (since their amount has changed)
- adapt `profile.py` based on the documentation for the columns

In my opinion, a failure with regards to Z3's path is better than some failure in `profile.py` due to the unexpected format, since in the former case I believe it is more likely for a user to learn about the version issue. However, it is no problem for me to change it, so I leave that decision up to you.

Reply:
Ok, Sounds good. I will merge this PR. can you please add this explanation on what to change to the README or to some issue on the issue tracker or some page on the wiki? It would be good to be able to find this easily later.